src/_pytest/assertion/util.py

```diff
@@ -423,6 +431,8 @@ def _compare_eq_cls(
         fields_to_check = [
             field.name for field in all_fields if getattr(field, ATTRS_EQ_FIELD)
         ]
+    elif isnamedtuple(left):
+        fields_to_check = left._fields
```

Comment:
moved from: https://github.com/pytest-dev/pytest/issues/7527#issuecomment-665050656

FWIW, I like the "._fields" approach but I think it should fallback to regular tuple comparison if the attributes are not exhausted.

Also I think we should just change the *display* of fields using _fields, and continue looking up items in the tuple by index/iteration

Reply:
> FWIW, I like the "._fields" approach but I think it should fallback to regular tuple comparison if the attributes are not exhausted.

Is this about checking the length of `_fields` and the object itself in the other comment?

> Also I think we should just change the display of fields using _fields, and continue looking up items in the tuple by index/iteration

Sorry, I don't get this. Is this about `_fields` being potentially used for some other purpose in custom subclass of tuple and to rely on iterating the namedtuple by index?